### PR TITLE
Removes deprecated expires and deadline fields

### PR DESCRIPTION
**Pull request by kritisingh1**
Reference : Bug 1437552
Removes `deadline` and `expires` fields from taskcluster-tools

### CODE REVIEW

File: src/utils.js [hunk 1]
```diff
@@ -78,9 +78,7 @@ export const parameterizeTask = task =>
     ),
     {
       retries: 0,
-      deadline: fromNowJSON('12 hours'),
       created: fromNowJSON(),
-      expires: fromNowJSON('7 days'),
```
Comment (djmitche) [inline src/utils.js]: These two will need to stay -- *tasks* still have deadline and expires -- we're just removing them from *hooks*.

File: src/views/HooksManager/HookEditor.js [hunk 2]
```diff
@@ -17,8 +17,6 @@ const initialHook = {
     emailOnError: true
   },
   schedule: [],
-  expires: '3 months',
-  deadline: '6 hours',
   task: {
     provisionerId: 'aws-provisioner-v1',
```
Comment (djmitche) [inline src/views/HooksManager/HookEditor.js]: Let's add and expires and deadline here, using JSON-e's $fromNow.

File: src/views/TaskCreator/TaskCreator.js [hunk 3]
```diff
@@ -14,9 +14,6 @@ const defaultTask = {
   provisionerId: 'aws-provisioner-v1',
   workerType: 'tutorial',
   created: moment().toISOString(),
-  deadline: moment()
-    .add(3, 'hours')
-    .toISOString(),
```
Comment (djmitche) [inline src/views/TaskCreator/TaskCreator.js]: This change is not related to hooks, so please revert :)

File: src/views/UnifiedInspector/ActionsMenu.js [hunk 4]
```diff
@@ -142,14 +142,11 @@ export default class ActionsMenu extends React.PureComponent {
...
     const taskId = nice();
     const now = Date.now();
-    const created = Date.parse(task.created);
 
     await queue.createTask(
       taskId,
       merge(omit(['dependencies'], clone(task)), {
         retries: 0,
-        deadline: new Date(now + Date.parse(task.deadline) - created).toJSON(),
-        expires: new Date(now + Date.parse(task.expires) - created).toJSON(),
```
Comment (djmitche) [inline src/views/UnifiedInspector/ActionsMenu.js]: same (not about hooks)

File: src/views/UnifiedInspector/TaskDetails.js [hunk 5]
```diff
@@ -163,20 +163,6 @@ export default class TaskDetails extends React.PureComponent {
...
-              </td>
-            </tr>
-
-            <tr>
-              <td>Expires</td>
-              <td>
-                <DateView date={task.expires} />
-              </td>
-            </tr>
-
```
Comment (djmitche) [inline src/views/UnifiedInspector/TaskDetails.js]: same

File: src/views/HooksManager/HookEditor.js (hunk 2 above)
Comment (kritisingh1) [reply src/views/HooksManager/HookEditor.js]: I need to add the two fields in the `task : {..}` and not the place from where they are removed, right? @djmitche

File: src/views/HooksManager/HookEditor.js [hunk 6]
```diff
@@ -17,8 +17,8 @@ const initialHook = {
     emailOnError: true
   },
   schedule: [],
-  expires: '3 months',
-  deadline: '6 hours',
+  expires: { $fromNow: '3 months' },
```
Comment (eliperelman) [inline src/views/HooksManager/HookEditor.js]: According to the [hooks schema](http://schemas.taskcluster.net/hooks/v1/create-hook-request.json), these fields are deprecated and it says to define these in the task template, which if I am correct, go in the `task.expires` and `task.deadline` properties.

@djmitche am I reading this correctly?

Comment (djmitche) [reply src/views/HooksManager/HookEditor.js]: Oh, yes! Sorry!

Comment (eliperelman) [reply src/views/HooksManager/HookEditor.js]: OK whew!

Comment (kritisingh1) [reply src/views/HooksManager/HookEditor.js]: @djmitche So I need to remove those two lines, right?

**Comment by kritisingh1** [comment]
@djmitche I have removed them from hooks. Do I need to add them to tasks or leave them as it is since task is now an object? Please  review and let me know the changes required. Thanks

**Comment by djmitche** [comment]
Yes, they need to be added in the `task` portion of the hook, just below where you removed them from.

**Comment by eliperelman** [comment]
Hello! I recently committed a large change to this repo, and your PR will need to be rebased. If you need a guide on how to do a rebase, have a read of:

http://blessing.io/git/git-rebase/open-source/2015/08/23/git-rebasing-to-resolve-conflicts.html
